src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -383,6 +384,78 @@ public PaginatedResult getVulnerabilities(Component component, boolean includeSu
         return result;
     }
 
+    /**
+     * Returns vulnerabilities by source and vulnerability IDs.
+     * Most efficient for fetching multiple vulnerabilities with known sources.
+     * @param sourceToVulnIds map of source to set of vulnerability IDs
+     * @param includeVulnerableSoftware whether to include vulnerable software in fetch plan
+     * @return list of matching Vulnerability objects
+     */
+    public List<Vulnerability> getVulnerabilitiesBySourceAndVulnIds(
+            Map<String, Set<String>> sourceToVulnIds,
+            boolean includeVulnerableSoftware) {
+
+        if (sourceToVulnIds == null || sourceToVulnIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        final Query<Vulnerability> query = pm.newQuery(Vulnerability.class);
+        query.getFetchPlan().addGroup(Vulnerability.FetchGroup.COMPONENTS.name());
+        if (includeVulnerableSoftware) {
+            query.getFetchPlan().addGroup(Vulnerability.FetchGroup.VULNERABLE_SOFTWARE.name());
+        }
+
+        StringBuilder filter = new StringBuilder();
+        Map<String, Object> paramValues = new HashMap<>();
+
+        int idx = 0;
+        for (Map.Entry<String, Set<String>> entry : sourceToVulnIds.entrySet()) {
+            if (entry.getValue() == null || entry.getValue().isEmpty()) {
+                continue;
+            }
+
+            if (!filter.isEmpty()) {
+                filter.append(" || ");
+            }
+
+            String sourceParam = "source" + idx;
+            String idsParam = "vulnId" + idx;
+
+            // (source == :source0 && :vulnId0.contains(vulnId))
+            filter.append("(source == :").append(sourceParam)
+                    .append(" && :").append(idsParam).append(".contains(vulnId))");
+
+            paramValues.put(sourceParam, entry.getKey());
+            paramValues.put(idsParam, new ArrayList<>(entry.getValue()));
+
+            idx++;
+        }
+
+        if (filter.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        query.setFilter(filter.toString());
+
+        @SuppressWarnings("unchecked")
+        List<Vulnerability> fetched = (List<Vulnerability>) query.executeWithMap(paramValues);
+
+        if (fetched == null || fetched.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        @SuppressWarnings("unchecked")
+        List<Vulnerability> detached = (List<Vulnerability>) pm.detachCopyAll(fetched);
+
+        // Safely apply aliases
+        for (Vulnerability vuln : detached) {
+            vuln.setAliases(getVulnerabilityAliases(vuln));
+        }
```

Comment:
Aliases are not required for the Trivy analysis task. Detaching has a negative effect for the Trivy task because it then has to immediately attach the records back in order to perform persistence operations (e.g., adding a component) with them.